hamilton/function_modifiers/expanders.py

```diff
@@ -759,13 +759,25 @@ def validate(self, fn: Callable):
         if typing_inspect.is_generic_type(output_type):
             base_type = typing_inspect.get_origin(output_type)
             if base_type == dict or base_type == Dict:
-                pass
+                _validate_extract_fields(self.fields)
             else:
                 raise base.InvalidDecoratorException(
                     f"For extracting fields, output type must be a dict or typing.Dict, not: {output_type}"
                 )
         elif output_type == dict:
-            pass
+            _validate_extract_fields(self.fields)
+        elif typing_extensions.is_typeddict(output_type):
+            if self.fields is None:
+                self.fields = typing.get_type_hints(output_type)
+            else:
+                # check that fields is a subset of TypedDict that is defined
+                typed_dict_fields = typing.get_type_hints(output_type)
+                for k, v in self.fields.items():
+                    if typed_dict_fields.get(k, None) != v:
```

Comment:
Subtle issue here --  if the `typed_dict_fields.get(k)` is a subclass of `self.fields[k]` it should work (I think, although typing can be confusing). Use `custom_subclass_check`?

Reply:
What would the use case be? I think someone using a subclass type of the field type in a typeddict is stretching things... can always extend, but for now I don't think we need to support.

Reply:
The use-case is if the fields declare a `Sequence` and the typed dict is a list. It's valid. This is an issue that we've had with the rest of our codebase (not accounting for subclasses) that shows up in weird ways. 

- pandas versus geopandas (I think)
- Sequence versus list
- Mapping versus dict, etc...

As to why you'd want this, this will happen if you're using a typed dict from somewhere else. While it's unlikely, we do this everywhere we can in the codebase.

We have a utility function that has been tuned to get this right: https://github.com/DAGWorks-Inc/hamilton/blob/4e8dc43ca802b9fcdf9059dfdb67f68d5cc38e41/hamilton/htypes.py#L38.

Reply:
Because we already have that function I'm happy to delegate to it.

Reply:
Yeah would not be worth it if we hadn't commonly be doing it elsewhere.